tests/test_linkml_files.py

```diff
@@ -1,64 +1,196 @@
-
-
-class LinkMLFilesTestCase(unittest.TestCase):
-    """ Test that linkml_model/linkml_files.py work """
-    def test_basic_rules(self):
-        self.assertEqual("https://w3id.org/linkml/annotations.yaml",
-                         URL_FOR(Source.ANNOTATIONS, Format.YAML))
-        self.assertEqual("https://w3id.org/linkml/meta.model.context.jsonld",
-                         URL_FOR(Source.META, Format.NATIVE_JSONLD))
-        self.assertEqual(os.path.join(root_path, "model/schema/meta.yaml"),
-                         LOCAL_PATH_FOR(Source.META, Format.YAML))
-        print(LOCAL_PATH_FOR(Source.META, Format.YAML))
-        self.assertEqual(os.path.join(root_path, "jsonld/types.model.context.jsonld"),
-                         LOCAL_PATH_FOR(Source.TYPES, Format.NATIVE_JSONLD))
-        self.assertEqual("https://linkml.github.io/linkml-model/model/schema/meta.yaml",
-                         GITHUB_IO_PATH_FOR(Source.META, Format.YAML))
-        self.assertEqual("https://linkml.github.io/linkml-model/jsonld/types.model.context.jsonld",
-                         GITHUB_IO_PATH_FOR(Source.TYPES, Format.NATIVE_JSONLD))
-        self.assertEqual("https://raw.githubusercontent.com/linkml/linkml-model/main/jsonld/meta.model.context.jsonld",
-                         GITHUB_PATH_FOR(Source.META, Format.NATIVE_JSONLD, ReleaseTag.LATEST))
-        self.assertEqual("https://raw.githubusercontent.com/linkml/linkml-model/testing_branch/owl/mappings.owl.ttl",
-                         GITHUB_PATH_FOR(Source.MAPPINGS, Format.OWL, branch="testing_branch"))
-
-    @unittest.skipIf(SKIP_GITHUB_API, "Github API tests skipped")
-    def test_github_specific_rules(self):
-        """
-        Test accesses that require github API to access
+
+def test_basic_rules():
+    assert "https://w3id.org/linkml/annotations.yaml" == URL_FOR(Source.ANNOTATIONS, Format.YAML)
+    assert "https://w3id.org/linkml/meta.context.jsonld" == URL_FOR(Source.META, Format.NATIVE_JSONLD)
+    assert os.path.join(root_path, "model/schema/meta.yaml") == LOCAL_PATH_FOR(Source.META, Format.YAML)
+    print(LOCAL_PATH_FOR(Source.META, Format.YAML))
+    assert os.path.join(root_path, "jsonld/types.context.jsonld") == LOCAL_PATH_FOR(Source.TYPES, Format.NATIVE_JSONLD)
+    assert "https://linkml.github.io/linkml-model/latest/linkml_model/model/schema/meta.yaml" == GITHUB_IO_PATH_FOR(
+        Source.META, Format.YAML)
+    assert "https://linkml.github.io/linkml-model/latest/linkml_model/jsonld/types.context.jsonld" == \
+           GITHUB_IO_PATH_FOR(Source.TYPES, Format.NATIVE_JSONLD)
+    assert "https://raw.githubusercontent.com/linkml/linkml-model/main/linkml_model/jsonld/meta.context.jsonld" == \
+           GITHUB_PATH_FOR(Source.META, Format.NATIVE_JSONLD, ReleaseTag.LATEST)
+    assert "https://raw.githubusercontent.com/linkml/linkml-model/testing_branch/linkml_model/owl/mappings.owl.ttl" == \
+           GITHUB_PATH_FOR(Source.MAPPINGS, Format.OWL, branch="testing_branch")
```

Comment:
`test_basic_rules` is basically the same thing as `test_fixed_meta_url` - in general i only test fixed values when there is a specific reason for it to be fixed forever, so that's why i only tested those two URLs, but i think if we're going to test the rest of these then they should be in a test structure that reflects the meaning of the test :)

test each of these functions (`URL_FOR`, `LOCAL_PATH_FOR`, `GITHUB_IO_PATH_FOR` and `GITHUB_PATH_FOR`) separately, and some of what we want to test for might already be covered by other tests.
- `URL_FOR`, `LOCAL_PATH_FOR`, and `GITHUB_IO_PATH_FOR` are trivial string formatting functions, so they can be covered by unit-style tests. 
- `LOCAL_PATH_FOR` has a `test_local_paths` test for existence for all the files in the expected formats already.
- `URL_FOR` has `test_url_for_format` that tests the resolution of the generated URLs for the formats that have rewrite rules in the w3id repo. 
- `GITHUB_PATH_FOR` is substantially different than the others and should be tested separately for sure, 

ideally these would have docstrings about why they should be fixed forever (if they were just fixed forever, why wouldn't we just hardcode the strings in the module itself and call it a day?), but idrc about that ;)